tools/misc/split_coco.py

```diff
@@ -0,0 +1,96 @@
+import argparse
+import json
+import os
+
+import numpy as np
+
+
+def split_coco(data_root, fold, percent):
+    """Split COCO data for Semi-supervised object detection.
+
+    Args:
+        data_root: root of dataset
+        fold: the fold of dataset and set as random seed for data split
+        percent: percentage of labeled data
+    """
+
+    def save_anno(name, images, annotations):
+        print(
+            'Starting to split data {}.json saved ({} images {} annotations)'.
+            format(name, len(images), len(annotations)))
+        sub_annos = {}
+        sub_annos['images'] = images
+        sub_annos['annotations'] = annotations
+        sub_annos['licenses'] = annos['licenses']
+        sub_annos['categories'] = annos['categories']
+        sub_annos['info'] = annos['info']
+        path = '/'.join(ann_file.split('/')[:-3] + ['coco_semi_annos'])
+        if not os.path.exists(path):
+            os.mkdir(path)
+
+        with open('{}/{}.json'.format(path, name), 'w') as f:
+            json.dump(sub_annos, f)
+
+        print(
+            'Finishing to split data {}.json saved ({} images {} annotations)'.
+            format(name, len(images), len(annotations)))
+
+    np.random.seed(fold)
+    ann_file = data_root + 'annotations/instances_train2017.json'
+    with open(ann_file) as f:
+        annos = json.load(f)
```

Comment:
If we can use os.path to handle the path, which is safe in both linux and windows.